frontend/src/metabase/lib/embed.js

```diff
@@ -41,12 +40,24 @@ export function initializeEmbedding(store) {
         }
       }
     });
-    store.dispatch(
-      setOptions({
-        ...parseSearchOptions(window.location.search),
-        ...parseHashOptions(window.location.hash),
-      }),
-    );
+
+    if (IS_EMBED_PREVIEW) {
+      let currentLocation;
+      store.subscribe(() => {
+        const previousLocation = currentLocation;
+        currentLocation = store.getState().routing.locationBeforeTransitions;
+
+        if (
+          previousLocation &&
+          (currentLocation.search !== previousLocation.search ||
```

Comment:
It looks dangerous to skip certain updates based on this heuristic. Wdyt about using something more stable? I'm not sure, but maybe even checking for `pathname` is better